### ISSUE TITLE
Examples bug when filename has two periods

### ISSUE THREAD

**Issue by andreped** [gradio-app/gradio#4424]
### Describe the bug

Just observed that when attempting to use add an example to a web app with two periods, e.g., `test_data.nii.gz`, the updated filename after example click event corrupted the file extension.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

There is likely a much simpler way to reproduce it, but below is an example that somewhat mimics my application. I am trying to connect the output of the Example click event to be linked to a `gr.File()` widget.

```
import gradio as gr

with gr.Blocks() as demo:
    file_output = gr.File(file_count="single")
    file_output.upload(upload_func(), file_output, file_output)
    
    run_button = gr.Button("Run analysis")
    run_button.click(fn=run_func(), inputs=file_output, outputs=[...])
    
    gr.Examples(examples=["test_data.nii.gz"], inputs=[...], outputs=[...],fn=upload_func(), cache_examples=True)
```

### Screenshot

See updated filename after attempting to use the Example with extension `.nii.gz`.

![Screenshot 2023-06-05 at 17 29 25](https://github.com/gradio-app/gradio/assets/29090665/0ea5f6ac-f9ea-42b2-873c-df2cfa7bdc24)

### Logs

```shell
Not necessary.
```

### System Info

```shell
gradio version: 3.32.0
Browser: Google Chrome
Operating system: Ubuntu Linux 20.04

gradio app deployed through Docker on Hugging Face spaces

The [neukit](https://huggingface.co/spaces/andreped/neukit) gradio app itself can be accessed [here](https://huggingface.co/spaces/andreped/neukit).
```

### Severity

serious, but I can work around it

**Comment by abidlabs** [comment]
@freddyaboulton @andreped it looks like this issue happens because we keep only the final suffix as the file extension for the temporary copy of the file that we create. In almost every case, this is the right thing to do, as filenames can have multiple periods (".") in the filename, but only the final suffix corresponds to the file extension. For example, if the filename is "a.foo.mp4", the file extension should be ".mp4". 

However, it looks like there are some file extensions (like `.nii.gz` or `tar.gz`) where this leads to the file becoming corrupted, or at least not being recognized as the correct filetype. I think the safer approach is actually to keep *all* of the suffixes. So for example:

"a.foo.mp4" --> "temp_name.foo.mp4" 
"a.nii.gz" --> "temp_name.nii.gz"

**Comment by freddyaboulton** [comment]
Thanks for looking into that @abidlabs ! This only happens for cached examples. I have a fix in #4440 - will add a unit test when I'm back home later and open it up for review.
